### PR TITLE
`std.Target`: Remove more dead OS/architecture tags

### PR DESCRIPTION
**Pull request by alexrp**
See commit messages.

### CODE REVIEW

**Comment by rohlem** [comment]
> > Our current policy is that hobby OSs should use the `other` tag.

> Can you link to anything supporting that, and a rough definition for "hobby OS"? I think Haiku is widely regarded as one, and I have a feeling that the policy is actually looking at _popularity_ :^)

I agree that some definitive wording to point to would be nice.
I personally think that, besides "popularity", activeness and collaboration are maybe more important factors.
I remember there being a couple of PRs specifically fixing and adding functionality for Haiku to `std`, and there are still many mentions of the `.haiku` tag in `std`.
Maybe that effort could be externalized into a separate package using the bring-your-own-OS mechanism, but before those mechanisms are finalized it makes some sense to have it upstream.

In comparison, `.ananas` seems to not have any specialized code paths anywhere, and as the commit message points out, even their upstream repo hasn't had activity in the last 2 years, so it's unlikely we'll see contributors wanting to fix `std`, nor use it as a field test for the BYO-OS design.

**Comment by alexrp** [comment]
> Can you link to anything supporting that, and a rough definition for "hobby OS"? I think Haiku is widely regarded as one, and I have a feeling that the policy is actually looking at _popularity_ :^)

I do not know that we have an actual, objective definition. But see #3784.

I think Ananas is an open-and-shut case, though. It's written in C, development has stalled for 2 years, and it only has 55 stars on GitHub. I think most would agree that it falls squarely in any reasonable definition of "hobby OS". It was only ever included in `std.Target` because LLVM for some reason included it in its `TargetParser` at one point (but no longer). OTOH, Haiku seems to be actively developed and it seems to have an actual user base.

For comparison, like Haiku, I didn't remove [Hermit](https://github.com/hermit-os) either. I think both of these have enough of a following that "hobby OS" is no longer an appropriate moniker.

This is just my thinking, though; I don't know what Andrew considers a "hobby OS" in the context of #3784.

**Comment by Vexu** [comment]
Are `Arch`, `Abi` and `Os.Tag` sorted in any way? If not could you sort them alongside one of your other PRs? It'd at least be nice to have `ps3` next to 4 & 5. `none` and `other` should be kept separate though.

**Comment by andrewrk** [comment]
Regarding "other": this tag is intended to facilitate the use case when no other tag is appropriate. Maybe your OS did get added to the zig std lib but you're stuck on an old zig version for example. The policy about adding OS tags is that any active project is a candidate. If the project looks dead, it's fair game to remove; if the project looks alive, it's fair game to add. Old consoles should be kept because these are permanently alive thanks to emulator communities.

In short summary, if any OS tag is desired, then it has a home in the std lib. Deleting a "dead" CPU arch or OS should never be controversial.

And it's perfectly fine to sort them.
